### ISSUE TITLE
support attaching attestation right after generate it

### ISSUE THREAD

**Issue by developer-guy** [anchore/syft#990]
**What would you like to be added**:

While reading the [blog post](https://anchore.com/sbom/creating-sbom-attestations-using-syft-and-sigstore/) on Anchore's website about generating attestations with Syft (really excellent blog post, btw), We've noticed that to attach attestations to the image Syft requires additional cosign command, so, we can provide the same functionality in Syft without requiring other cosign command by simply adding a new flag named --attach or --load, etc.

**Why is this needed**:

Eliminating the second command to attach attestation to the image

**Additional context**:
- We're willing to do this :point_up_2:
- Cross-ref-issue: https://github.com/goreleaser/goreleaser/issues/2997

cc: @dentrax

**Comment by Dentrax** [comment]
+1, this one very straightforward task that we can handle this in [Run()](https://github.com/anchore/syft/blob/d2d532f4a852067c9ccba2329a8105ac5479e2ae/cmd/syft/cli/attest/attest.go#L60) function. Eventually UX will be:

```diff
- $ syft attest --key ./cosign.key <my-image> -o cyclonedx-json > ./my-image-sbom.att.json
- $ cosign attach attestation <my-image> --attestation ./my-image-sbom.att.json
vs.
+ $ syft attest --attach --key ./cosign.key <my-image> -o cyclonedx-json
```

**Comment by developer-guy** [comment]
Fixed by: #910

**Comment by developer-guy** [comment]
![Screen Shot 2022-05-11 at 22 20 32](https://user-images.githubusercontent.com/16693043/167929017-ad4644a7-5f5a-4a7c-839d-18a0dbda9bc6.png)
